### PR TITLE
Allow serving multiple channels with one instance

### PR DESCRIPTION
**Pull request by alga**
Enabled with another env var: `IRCLOG_CHAN_DIR`.

### CODE REVIEW

File: src/irclog2html/irclogserver.py [hunk 1]
```diff
@@ -22,26 +22,59 @@
...
 from .irclog2html import CSS_FILE, LogParser
 from .irclogsearch import (
     DEFAULT_LOGFILE_PATH, DEFAULT_LOGFILE_PATTERN, search_page,
+    HEADER, FOOTER
 )
 
 
-def get_path(environ):
+def dir_listing(stream, path):
+    """Primitive listing of  subdirectories"""
```
Comment (mgedmin) [inline src/irclog2html/irclogserver.py]: Two spaces after 'of'.

File: src/irclog2html/irclogserver.py [hunk 2]
```diff
@@ -22,26 +22,59 @@
...
+    print(HEADER, file=stream)
+    print(u"<h1>IRC logs</h1>", file=stream)
+    print(u"<ul>", file=stream)
+    for name in os.listdir(path):
```
Comment (mgedmin) [inline src/irclog2html/irclogserver.py]: `sorted(os.listdir(path))`.

File: src/irclog2html/irclogserver.py [hunk 3]
```diff
@@ -22,26 +22,59 @@
...
+        if os.path.isdir(os.path.join(path, name)):
+            print(u'<li><a href="%s/">%s</a></li>'
+                  % (quote_plus(name), name),
```
Comment (mgedmin) [inline src/irclog2html/irclogserver.py]: also escape() maybe?

**Comment by mgedmin** [comment]
The Travis build failure is very mysterious.  Any ideas?

**Comment by mgedmin** [comment]
It's probably the release of [coverage 4.0](https://pypi.python.org/pypi/coverage/4.0).

**Comment by mgedmin** [comment]
Travis fixed in trunk.

If you add some commits (e.g. fixing the sorted() and escape() bits, and the docstring), GitHub ought to merge and Travis ought to succeed.

**Comment by alga** [comment]
Beautiful quotes were breaking tox:
```py33 create: /home/alga/src/irclog2html/.tox/py33
ERROR: InterpreterNotFound: python3.3
py34 inst-nodeps: /home/alga/src/irclog2html/.tox/dist/irclog2html-2.15.0.dev0.zip
ERROR: invocation failed (exit code 2), logfile: /home/alga/src/irclog2html/.tox/py34/log/py34-11.log
ERROR: actionid: py34
msg: installpkg
cmdargs: [local('/home/alga/src/irclog2html/.tox/py34/bin/pip'), 'install', '-U', '--no-deps', '/home/alga/src/irclog2html/.tox/dist/irclog2html-2.15.0.dev0.zip']
env: {'PATH': '/home/alga/src/irclog2html/.tox/py34/bin:/home/alga/.local/bin:/home/alga/bin:/home/alga/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games', 'VIRTUAL_ENV': '/home/alga/src/irclog2html/.tox/py34', 'PYTHONHASHSEED': '528825230'}

You are using pip version 7.0.1, however version 7.1.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
DEPRECATION: --download-cache has been deprecated and will be removed in the future. Pip now automatically uses and configures its cache.
Processing ./.tox/dist/irclog2html-2.15.0.dev0.zip
Building wheels for collected packages: irclog2html
  Running setup.py bdist_wheel for irclog2html
  Stored in directory: /home/alga/.cache/pip/wheels/ca/ed/b0/f0cca3a5d76814bb63b8cdca71202ac66341878cfd036acfa2
Successfully built irclog2html
Installing collected packages: irclog2html
  Found existing installation: irclog2html 2.14.1.dev0
    Uninstalling irclog2html-2.14.1.dev0:
      Successfully uninstalled irclog2html-2.14.1.dev0
  Rolling back uninstall of irclog2html
Exception:
Traceback (most recent call last):
  File "/home/alga/src/irclog2html/.tox/py34/lib/python3.4/site-packages/pip/basecommand.py", line 223, in main
    status = self.run(options, args)
  File "/home/alga/src/irclog2html/.tox/py34/lib/python3.4/site-packages/pip/commands/install.py", line 297, in run
    root=options.root_path,
  File "/home/alga/src/irclog2html/.tox/py34/lib/python3.4/site-packages/pip/req/req_set.py", line 622, in install
    **kwargs
  File "/home/alga/src/irclog2html/.tox/py34/lib/python3.4/site-packages/pip/req/req_install.py", line 808, in install
    self.move_wheel_files(self.source_dir, root=root)
  File "/home/alga/src/irclog2html/.tox/py34/lib/python3.4/site-packages/pip/req/req_install.py", line 1003, in move_wheel_files
    isolated=self.isolated,
  File "/home/alga/src/irclog2html/.tox/py34/lib/python3.4/site-packages/pip/wheel.py", line 241, in move_wheel_files
    name, user=user, home=home, root=root, isolated=isolated
  File "/home/alga/src/irclog2html/.tox/py34/lib/python3.4/site-packages/pip/locations.py", line 182, in distutils_scheme
    d.parse_config_files()
  File "/usr/lib/python3.4/distutils/dist.py", line 394, in parse_config_files
    parser.read(filename)
  File "/usr/lib/python3.4/configparser.py", line 672, in read
    self._read(fp, filename)
  File "/usr/lib/python3.4/configparser.py", line 993, in _read
    for lineno, line in enumerate(fp, start=1):
  File "/home/alga/src/irclog2html/.tox/py34/lib/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 350: ordinal not in range(128)


py34 installed: irclog2html==2.14.1.dev0,mock==1.3.0,pbr==1.8.0,six==1.9.0,wheel==0.24.0,zope.exceptions==4.0.8,zope.interface==4.1.2,zope.testing==4.5.0,zope.testrunner==4.4.9
pypy inst-nodeps: /home/alga/src/irclog2html/.tox/dist/irclog2html-2.15.0.dev0.zip
pypy installed: cffi==0.8.6,funcsigs==0.4,greenlet==0.4.5,irclog2html==2.15.0.dev0,mock==1.3.0,pbr==1.8.0,readline==6.2.4.1,six==1.9.0,wheel==0.24.0,zope.exceptions==4.0.8,zope.interface==4.1.2,zope.testing==4.5.0,zope.testrunner==4.4.9
pypy runtests: PYTHONHASHSEED='528825230'
pypy runtests: commands[0] | zope-testrunner --test-path=src -v
Running tests at level 1
Running zope.testrunner.layer.UnitTests tests:
  Set up zope.testrunner.layer.UnitTests in 0.000 seconds.
  Running:
..........................................................................
  Ran 74 tests with 0 failures, 0 errors and 0 skipped in 0.188 seconds.
Tearing down left over layers:
  Tear down zope.testrunner.layer.UnitTests in 0.000 seconds.
ERROR:   py34: InvocationError: /home/alga/src/irclog2html/.tox/py34/bin/pip install -U --no-deps /home/alga/src/irclog2html/.tox/dist/irclog2html-2.15.0.dev0.zip (see /home/alga/src/irclog2html/.tox/py34/log/py34-11.log)
```

**Comment by mgedmin** [comment]
Oh, wow.  Was your locale ASCII-only?

**Comment by mgedmin** [comment]
Thank you!

**Pull request merged**